d3log/src/main.rs

```diff
@@ -0,0 +1,57 @@
+mod json_framer;
+mod batch;
+mod tcp_network;
+mod child;
+mod transact;
+
+
+use crate::{batch::Batch,
+            child::start_children,
+            typedefs::matrix::Matrix};
+
+use differential_datalog::ddval::DDValConvert;
+use std::str;
+use rustop::opts;
+use mm_ddlog::*;
+use differential_datalog::DeltaMap;
+use differential_datalog::{D3logLocationId};
+
+type Node = D3logLocationId;
+
+use std::convert::TryFrom;
+fn matrix(mat:Vec::<Vec::<u64>>) -> Result<Batch, String> {
+    let mut b = Batch::new(DeltaMap::new());
+    let mut i = 0;
+    let relid = Relations::try_from("matrix::Matrix").map_err(|_| format!("Unknown relation {}", "Matrix"))?;
+
+    for c in mat.iter() {
+        let mut j = 0;
+        for v in c.iter() {
+            b.insert(relid as usize, Matrix{i, j, v:(*v) as u32}.into_ddvalue(), 1);
```

Comment:
```suggestion
            b.insert(relid as usize, Matrix{i, j, v: v as u32}.into_ddvalue(), 1);
```

Reply:
|                     v: v as u32,
   |                        -^^^^^^^
   |                        |
   |                        cannot cast `&u64` as `u32`
   |                        help: dereference the expression: `*v`